android/build.gradle

```diff
@@ -15,7 +15,39 @@ buildscript {
     }
 }
 
-// This looks funny but it's necessary to keep backwards compatibility (: 
+task checkKotlinVersion {
+    doLast {
+        def kotlinVersion = rootProject.ext.has('kotlinVersion') ? rootProject.ext.get('kotlinVersion') : project.properties['RNVideo_kotlinVersion']
+        def requiredKotlinVersion = project.properties['RNVideo_requiredKotlinVersion']
+
+        // Function to compare version strings
+        def isVersionAtLeast = { version, requiredVersion ->
+            def (v1, v2) = [version, requiredVersion].collect { it.tokenize('.')*.toInteger() }
+            for (int i = 0; i < Math.max(v1.size(), v2.size()); i++) {
+                int val1 = i < v1.size() ? v1[i] : 0
+                int val2 = i < v2.size() ? v2[i] : 0
+                if (val1 < val2) {
+                    return false
+                } else if (val1 > val2) {
+                    return true
+                }
+            }
+            return true
+        }
+
+        if (!isVersionAtLeast(kotlinVersion, requiredKotlinVersion)) {
+            throw new GradleException("Kotlin version mismatch: Project is using Kotlin version $kotlinVersion, but it must be at least $requiredKotlinVersion. Please update the Kotlin version.")
+        } else {
+            println("Kotlin version is correct: $kotlinVersion")
+        }
+    }
+}
+
+tasks.matching { it.name.startsWith('pre') || it.name == 'compileDebugKotlin' }.all {
+    dependsOn checkKotlinVersion
+}
```

Comment:
I don't think that we need to do it that we - we can convert `checkKotlinVersion` task to function and call this function eg. in buildscript or somewhere else

Reply:
It's refactored.